src/Sarif/Core/FileData.cs

```diff
@@ -12,19 +15,44 @@ namespace Microsoft.CodeAnalysis.Sarif
     /// </summary>
     public partial class FileData : ISarifNode
     {
-        public static FileData Create(Uri uri, bool computeHashes)
+        public static FileData Create(Uri uri, SarifWriters.LoggingOptions loggingOptions, string mimeType = null, Encoding encoding = null)
         {
             if (uri == null) { throw new ArgumentNullException(nameof(uri)); }
 
+            mimeType = mimeType ?? SarifWriters.MimeType.DetermineFromFileExtension(uri);
+            encoding = encoding ?? Encoding.UTF8;
+
             var fileData = new FileData()
             {
-                MimeType = SarifWriters.MimeType.DetermineFromFileExtension(uri)
+                MimeType = mimeType,
+                Uri = uri
             };
 
-            if (computeHashes && uri.IsAbsoluteUri && uri.IsFile)
+            // Attempt to persist file contents and/or compute file hash and persist
+            // this information to the log file. In the event that there is some issue
+            // accessing the file, for example, due to ACLs applied to a directory,
+            // we currently swallow these exceptions without populating any requested
+            // data or putting a notification in the log file that a problem
+            // occurred. Something to discuss moving forward.
+            try
             {
-                HashData hashes = HashUtilities.ComputeHashes(uri.LocalPath);
-                fileData.Hashes = new List<Hash>
+                if (!uri.IsAbsoluteUri || !uri.IsFile || !File.Exists(uri.LocalPath))
+                {
+                    return fileData;
+                }
+
+                string filePath = uri.LocalPath;
+                bool encodeAsUtf8 = (fileData.MimeType != SarifWriters.MimeType.Binary);
```

Comment:
>fileData.MimeType != SarifWriters.MimeType.Binar [](start = 37, length = 48)

There are many MIME types other than Binary that are non-text. For example, application/zip.

Reply:
This is a very hard problem.

---
In reply to: [136173115](https://github.com/Microsoft/sarif-sdk/pull/612#discussion_r136173115) [](ancestors = 136173115)

Reply:
Well ok, first of all, this variable is no longer used, so omit this line.

---
In reply to: [136173223](https://github.com/Microsoft/sarif-sdk/pull/612#discussion_r136173223) [](ancestors = 136173223,136173115)